blockchain/store_test.go

```diff
@@ -0,0 +1,268 @@
+		_, _, err := doFn(func() (interface{}, error) {
+			db.Set(blockStoreKey, panicCauser)
+			_ = blockchain.NewBlockStore(db)
+			return nil, nil
+		})
+		assert.NotNil(t, err, "#%d panicCauser: %q expected a panic", i, panicCauser)
+		// Note: We can't check for the exact substring because
+		// recovered stack traces as of Sept 2017 seem to
+		// mask the message. See https://play.golang.org/p/W9NTc4Fstr
+		// TODO: Perhaps file an error on the Go project.
+	}
+
+	db.Set(blockStoreKey, nil)
+	bs = blockchain.NewBlockStore(db)
+	assert.Equal(t, bs.Height(), 0, "expecting nil bytes to be unmarshaled alright")
+}
+
+func TestBlockStoreGetReader(t *testing.T) {
+	db := db.NewMemDB()
+	// Initial setup
+	db.Set([]byte("Foo"), []byte("Bar"))
+	db.Set([]byte("Foo1"), nil)
+
+	bs := blockchain.NewBlockStore(db)
+
+	tests := [...]struct {
+		key  []byte
+		want []byte
+	}{
+		0: {key: []byte("Foo"), want: []byte("Bar")},
+		1: {key: []byte("KnoxNonExistent"), want: nil},
+		2: {key: []byte("Foo1"), want: nil},
+	}
+
+	for i, tt := range tests {
+		r := bs.GetReader(tt.key)
+		if r == nil {
+			assert.Nil(t, tt.want, "#%d: expected a non-nil reader", i)
+			continue
+		}
+		slurp, err := ioutil.ReadAll(r)
+		if err != nil {
+			t.Errorf("#%d: unexpected Read err: %v", i, err)
+		} else {
+			assert.Equal(t, slurp, tt.want, "#%d: mismatch", i)
+		}
+	}
+}
+
+func freshBlockStore() (*blockchain.BlockStore, db.DB) {
+	db := db.NewMemDB()
+	return blockchain.NewBlockStore(db), db
+}
+
+func TestBlockStoreSaveLoadBlock(t *testing.T) {
+	bs, _ := freshBlockStore()
+	noBlockHeights := []int{0, -1, 100, 1000, 2}
+	for i, height := range noBlockHeights {
+		if g := bs.LoadBlock(height); g != nil {
+			t.Errorf("#%d: height(%d) got a block; want nil", i, height)
+		}
+	}
+
+	// Setup, test data
+	part2 := &types.Part{Index: 1, Bytes: data.Bytes([]byte{
+		0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x01, 0x00, 0x00, 0x00, 0x01, 0x01, 0x01, 0x00,
+		0x00, 0x01, 0x0a, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+	})}
+	part1 := &types.Part{Index: 0, Bytes: data.Bytes([]byte{
+		0x01, 0x01, 0x01, 0x0a, 0x62, 0x6c, 0x6f, 0x63, 0x6b, 0x5f, 0x74, 0x65,
+		0x73, 0x74, 0x01, 0x01, 0xa1, 0xb2, 0x03, 0xeb, 0x3d, 0x1f, 0x44, 0x40, 0x01, 0x64, 0x00,
+	})}
+	validPartSet := types.NewPartSetFromHeader(types.PartSetHeader{Total: 2})
+	validPartSet.AddPart(part1, false)
+	validPartSet.AddPart(part2, false)
+
+	incompletePartSet := types.NewPartSetFromHeader(types.PartSetHeader{Total: 2})
+
+	uncontiguousPartSet := types.NewPartSetFromHeader(types.PartSetHeader{Total: 0})
+	uncontiguousPartSet.AddPart(part2, false)
+	seenCommit1 := &types.Commit{Precommits: []*types.Vote{{Height: 10}}}
+
+	// End of setup, test data
+
+	tuples := []struct {
+		block            *types.Block
+		parts            *types.PartSet
+		seenCommit       *types.Commit
+		wantErr          bool
+		wantPanic        bool
+		eraseDBForCommit bool
+		skipReason       string
```

Comment:
can't see where this is used

Reply:
Ahh crap, I forgot to remove it after the last commit, I'll update that, thanks.